### PR TITLE
BUG: Fix FileReaderRawBinaryWithHeader ignoring extended header size,…

### LINKED ISSUE #892: Read mrc file
We use the ``FileReaderMRC`` to read mrc file following the example in ``examples/tomo/data/mrc_io.py``. Just read the given file ``test.mrc``, no problem. But if we read the interesting mrc file, get the following error:

>Traceback (most recent call last):

>  File "<ipython-input-6-8bf8cb54b126>", line 1, in <module>
>    runfile('/home/chchen/odl/odl/deform/read_mrc_data.py', wdir='/home/chchen/odl/odl/deform')

>  File "/home/chchen/miniconda2/lib/python2.7/site-packages/spyder/utils/site/sitecustomize.py", line > 866, in runfile
>    execfile(filename, namespace)

>  File "/home/chchen/miniconda2/lib/python2.7/site-packages/spyder/utils/site/sitecustomize.py", line > 94, in execfile
>   builtins.execfile(filename, *where)

>  File "/home/chchen/odl/odl/deform/read_mrc_data.py", line 114, in <module>
>    header, data = reader.read()

>  File "/home/chchen/odl/odl/tomo/data/uncompr_bin.py", line 385, in read
>    return self.read_header(), self.read_data()

>  File "/home/chchen/odl/odl/tomo/data/mrc.py", line 632, in read_data
>    order='F')

>ValueError: total size of new array must be unchanged
> > > >

### PR DESCRIPTION
**Pull request by adler-j**
… closes #892

### CODE REVIEW

**Comment by kohr-h** [comment]
I disagree with the fix. Extended header is a property of the MRC file format and has nothing to do with the general uncompressed binary reader.
What you call `total_header_size` is supposed to be `header_size` to begin with, since it makes no sense to call something "header size" if the actual header is larger. And that's actually what is supposed to happen already now, see [here](https://github.com/odlgroup/odl/blob/8c8070ddc5e330b511cee25a1cbcda651669902a/odl/tomo/data/mrc.py#L215-L232).
My guess is that the order of inheritance bites here, and the MRC reader needs to explicitly override the `header_size` property with the one from the mixin class.
That means that inserting a `header_size = MRCHeaderProperties.header_size` at [this](https://github.com/odlgroup/odl/blob/8c8070ddc5e330b511cee25a1cbcda651669902a/odl/tomo/data/mrc.py#L477) should fix the issue.

**Comment by adler-j** [comment]
Wouldn't this whole thing be fixed by changing the order of inheritance so that the properties thing comes first?

**Comment by kohr-h** [comment]
Yes that's right, in particular since the mixin class has no init, that shouldn't pose any difficulty.

**Comment by adler-j** [comment]
Prefer this way? Now everything seems to be working properly.

File: odl/tomo/data/mrc.py [hunk 1]
```diff
@@ -473,8 +473,6 @@ def __init__(self, file, header_fields=None):
 
     print_mrc2014_spec = staticmethod(print_mrc2014_spec)
```
Comment (adler-j) [inline odl/tomo/data/mrc.py]: Should this line be moved to `MRCHeaderProperties`?

Comment (kohr-h) [inline odl/tomo/data/mrc.py]: Yes why not, together with the next one.

**Comment by kohr-h** [comment]
> Prefer this way? Now everything seems to be working properly.

Much  better :-)

**Comment by adler-j** [comment]
Need an approval. Then merge.

**Comment by adler-j** [comment]
New issue on the test: #894

**Pull request merged**